src/EditorFeatures/Core/InheritanceMargin/AbstractInheritanceMarginService_Helpers.cs

```diff
@@ -0,0 +1,171 @@
+            ImmutableArray<ISymbol> implementedMembers,
+            ImmutableArray<ISymbol> overridenMembers,
+            ImmutableArray<ISymbol> overridingMembers,
+            CancellationToken cancellationToken)
+        {
+            var implementingMemberItems = await implementingMembers
+                .SelectAsArrayAsync(symbol => CreateInheritanceItemAsync(solution, symbol, InheritanceRelationship.Implementing, cancellationToken)).ConfigureAwait(false);
+            var implementedMemberItems = await implementedMembers
+                .SelectAsArrayAsync(symbol => CreateInheritanceItemAsync(solution, symbol, InheritanceRelationship.Implemented, cancellationToken)).ConfigureAwait(false);
+            var overridenMemberItems = await overridenMembers
+                .SelectAsArrayAsync(symbol => CreateInheritanceItemAsync(solution, symbol, InheritanceRelationship.Overriden, cancellationToken)).ConfigureAwait(false);
+            var overridingMemberItems = await overridingMembers
+                .SelectAsArrayAsync(symbol => CreateInheritanceItemAsync(solution, symbol, InheritanceRelationship.Overriding, cancellationToken)).ConfigureAwait(false);
+
+            return new InheritanceMemberItem(
+                lineNumber,
+                memberSymbol.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat),
+                memberSymbol.GetGlyph(),
+                implementingMemberItems.Concat(implementedMemberItems)
+                    .Concat(overridenMemberItems)
+                    .Concat(overridingMemberItems));
+        }
+
+        /// <summary>
+        /// For the <param name="memberSymbol"/>, get all the symbols implementing it.
+        /// </summary>
+        private static async Task<ImmutableArray<ISymbol>> GetImplementedSymbolsAsync(
+            Solution solution,
+            ISymbol memberSymbol,
+            CancellationToken cancellationToken)
+        {
+            if (memberSymbol is INamedTypeSymbol { IsSealed: false } namedTypeSymbol)
+            {
+                var derivedTypes = await GetDerivedTypesAndImplementationsAsync(solution, namedTypeSymbol, cancellationToken).ConfigureAwait(false);
+                return derivedTypes.OfType<ISymbol>().ToImmutableArray();
+            }
+
+            if (memberSymbol is IMethodSymbol or IEventSymbol or IPropertySymbol
+                 && memberSymbol.ContainingSymbol.IsInterfaceType())
+            {
+                return await SymbolFinder.FindMemberImplementationsArrayAsync(
+                    memberSymbol,
+                    solution,
+                    cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+
+            return ImmutableArray<ISymbol>.Empty;
+        }
+
+        /// <summary>
+        /// Get members overriding the <param name="memberSymbol"/>
+        /// </summary>
+        private static ImmutableArray<ISymbol> GetOverridingSymbols(ISymbol memberSymbol)
```

Comment:
```suggestion
        /// Get members overridden by the <param name="memberSymbol"/>
        /// </summary>
        private static ImmutableArray<ISymbol> GetOverriddenSymbols(ISymbol memberSymbol)
```

Reply:
I will put a full table for thoese 4 images to help understand